runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/Info.java

```diff
@@ -131,11 +78,53 @@ public Info build()
             Info info = new Info(path);
             if (readonly)
             {
-                info.readWorkerCount();
+                try
+                {
+                    byte[] bytes = Files.readAllBytes(info.info);
+                    ByteBuffer byteBuf = ByteBuffer
+                        .wrap(bytes, Long.BYTES, Integer.BYTES)
+                        .order(nativeOrder());
+                    info.workerCount = byteBuf.getInt();
+                }
+                catch (IOException ex)
+                {
+                    System.out.printf("Error: %s is not readable\n", info.info);
```

Comment:
We don't typically catch exceptions deep inside the engine, instead we let them propagate out so that we know there have been zero issues if the engine is still running.

What happens if you let this out via `LangUtil.rethrowUnchecked(ex)`?

Reply:
I added `LangUtil.rethrowUnchecked(ex)` in the catch block in both cases. Now the engine prints the error message and exits if the file can't be opened. Pls chk.